h2o-core/src/main/java/water/util/FileUtils.java

```diff
@@ -97,33 +101,74 @@ else if (file.isDirectory()) {
    *  @param fname filename
    *  @return      Found file or null */
   public static File locateFile(String fname) {
-    // When run from eclipse, the working directory is different.
-    // Try pointing at another likely place
+
     File file = new File(fname);
-    if( !file.exists() )
-      file = new File("target/" + fname);
-    if( !file.exists() )
+    if (file.exists()) return file;
+    
+    final Optional<File> fileInPredefinedPath = findFileInPredefinedPath(fname);
+    if (fileInPredefinedPath.isPresent()) return fileInPredefinedPath.get();
+    
+    file = new File("target/" + fname);
+    if (!file.exists())
       file = new File("../" + fname);
-    if( !file.exists() )
+    if (!file.exists())
       file = new File("../../" + fname);
-    if( !file.exists() )
+    if (!file.exists())
       file = new File("../../../" + fname);
-    if( !file.exists() )
+    if (!file.exists())
       file = new File("../target/" + fname);
-    if( !file.exists() )
+    if (!file.exists())
       file = new File(StringUtils.expandPath(fname));
-    if( !file.exists() )
+    if (!file.exists())
       file = null;
     return file;
   }
 
-  private static void check(boolean cond, String msg) throws IOException{
+  /**
+   * @param fileName File name/path to search for in pre-defined search path
+   * @return An {@link Optional} with the file inside, if the H2O_FILES_SEARCH_PATH is defined and the file exists.
+   * Otherwise an empty {@link Optional}. Never null.
+   */
+  private static Optional<File> findFileInPredefinedPath(final String fileName) {
+    Objects.requireNonNull(fileName);
+
+    final String smallDataPathPrefix = System.getenv("H2O_FILES_SEARCH_PATH");
+    if (smallDataPathPrefix == null) return Optional.empty();
+
+    final StringBuilder localizedFileNameBuilder = new StringBuilder(smallDataPathPrefix);
+
+    if (!smallDataPathPrefix.endsWith("/")) {
+      localizedFileNameBuilder.append('/');
+    }
+    
+    // If the file starts with {"./", ".\"} (or multiple instances of these), strip it.
```

Comment:
do you really need to strip these? I think that /path/./with/./dots/ is the equivalent to one withtout the /./ parts

Reply:
The true point is to strip `../` or windows `..\` and chains of these from the very beginning. Will pushed improved example. And not `/` only - as this indicates the path is absolute. The fact that it's able to make the resulting path and thus Frame keys a little bit more readable is a bonus. :grin:

Reply:
I'll also add tests for that. Good point.